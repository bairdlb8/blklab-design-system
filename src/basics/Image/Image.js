import React from 'react'
import PropTypes from 'prop-types'
import './Image.scss'

//TODOs:
//Generate responsive versions
//

function Image({ src = 'https://source.unsplash.com/daily', alt = '' }) {
  return (
    <picture className="ds-image">
      <img src={src} />
    </picture>
  )
}

Image.propTypes = {
  src: PropTypes.string,
  alt: PropTypes.string,
  objectFit: PropTypes.oneOf(
    Object.keys({
      cover: 'cover',
      contain: 'contain',
      fill: 'fill'
    })
  )
}

export default Image
