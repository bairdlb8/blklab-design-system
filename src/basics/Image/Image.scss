.ds-image {
  overflow: hidden;
  img {
    width: 100%;
    height: 100%;
    object-position: center;
    object-fit: cover;
    max-width: 100%;
    max-height: 100%;
  }
}
