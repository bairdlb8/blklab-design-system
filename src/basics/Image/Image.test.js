import { render } from '@testing-library/react';
import React from 'react';
import Image from './Image';

describe('Image', () => {
    const defaultProps = {};

    it('should render', () => {
        const props = {...defaultProps};
        const { asFragment, queryByText } = render(<Image {...props} />);

        expect(asFragment()).toMatchSnapshot();
        expect(queryByText('Image')).toBeTruthy();
    });
});
