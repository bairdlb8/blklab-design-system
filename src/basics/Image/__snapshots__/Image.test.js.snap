// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image should render 1`] = `
<DocumentFragment>
  Image
</DocumentFragment>
`;
